lib/puppet/provider/podman_secret/podman.rb

```diff
@@ -95,20 +90,17 @@ def execute_podman_command(args, capture_output: false)
         'XDG_RUNTIME_DIR' => "/run/user/#{user_info.uid}",
       }
 
-      result = Puppet::Util::Execution.execute(
+      Puppet::Util::Execution.execute(
         [command(:podman)] + args,
         uid: user_info.uid,
         gid: user_info.gid,
         custom_environment: env,
         cwd: user_info.dir,
         failonfail: true,
       )
-      capture_output ? result : nil
-    elsif capture_output
-      podman(*args)
     else
+      ::ENV['PODMAN_IGNORE_CGROUPSV1_WARNING'] = 'true'
```

Comment:
Took me a moment to reproduce this (for the reader at home: you need to boot EL9 with `systemd.unified_cgroup_hierarchy=0` to get that).

I wonder if this is "the right way". It hides this particular warning, but surely podman can issue others, breaking our parsing again?

The warning is (correctly) printed to `stderr`, so I think instead of implementing "hide that one warning", we should collect `stdout` and `stderr` separately, and only parse `stdout` into our secret (and show `stderr` in the error case?)